src/SI_Toolkit/Predictors/predictor_wrapper.py

```diff
@@ -0,0 +1,136 @@
+import os
+import yaml
+from copy import deepcopy as dcp
+
+# predictors config
+predictors_config = yaml.load(open(os.path.join('SI_Toolkit_ASF', 'config_predictors.yml'), 'r'), Loader=yaml.FullLoader)
+
+
+class PredictorWrapper:
+    def __init__(self):
+
+        self.horizon = None
+        self.batch_size = None
+
+        self.predictor = None
+
+        self.predictors_config = dcp(predictors_config['predictors'])  # TODO: This must be read only!
```

Comment:
I googled for this and a solution could be

```suggestion
        self.predictors_config = MappingProxyType(predictors_config['predictors'])
```

Reply:
`from types import MappingProxyType`

Reply:
@marcinpaluch1994  What do you think?

Reply:
Done! Thank you!